gg/layer.go

```diff
@@ -145,6 +145,40 @@ func (l LayerPaths) apply(p *Plot, sort bool) {
 	}, p.Data().Tables()})
 }
 
+// LayerArea shades the area between two columns with a polygon. It is
+// useful in conjunction with ggstat.AggMax and ggstat.AggMin for
+// drawing the extents of data.
+type LayerArea struct {
+	// X names the column that defines the input of each point. If
+	// this is empty, it defaults to the first column.
+	X string
+
+	// Upper and Lower name columns that define the vertical
+	// bounds of the shaded area. If either is "", it defaults to
+	// 0.
+	Upper, Lower string
```

Comment:
It does matter which is upper and which is lower; they need to not cross because otherwise the area will not be drawn correctly. (I guess it's the case that you could swap them, but *shrug*)